File: leetcode/medium/Q338_CountingBits.py
from typing import List

class Solution:
    def countBits(self, num: int) -> List[int]:
        answer = [0]
        while len(answer) <= num:
            tmp = []
            for i in answer:
                tmp.append(i+1)
            answer += tmp
        return answer[:num+1]
s = Solution()
print(s.countBits(2))

File: src/easy/Q338_CountingBits.java
package easy;

public class Q338_CountingBits {
    class Solution {
        public int[] countBits(int n) {
            int[] ans = new int[n+1];
            for (int i = 0; i < n+1; i++) {
                int t = i;
                while(t > 0){
                    ans[i] += (t%2);
                    t = t/2;
                }
            }
            return ans;
        }
    }
}
